virtcontainers/kata_agent.go

```diff
@@ -451,7 +451,7 @@ func cmdToKataProcess(cmd types.Cmd) (process *grpc.Process, err error) {
 	var gid uint32
 	if len(parsedUser) > 1 {
 		i, err = strconv.ParseUint(parsedUser[1], 10, grpcUserBits)
-		if err != nil {
+		if err != nil && if len(cmd.PrimaryGroup) == 0 {
```

Comment:
@tedyu Can you explain the reason for doing this?
We expect user to be of the format `uid` or `uid:gid`, so if we receive an unxpected format we rightfully error out.

Reply:
What if cmd. PrimaryGroup is also present and parses correctly ?

Reply:
I prefer erroring out here in the case where we have got incorrect input.
cc @jodh-intel @grahamwhaley